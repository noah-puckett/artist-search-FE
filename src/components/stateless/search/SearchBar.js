import React from 'react';
import PropTypes from 'prop-types';

function SearchBar({ searchBoxInput, onChange }) {
    return (<>
        <input name="searchBoxInput" value={searchBoxInput} onChange={onChange} />
    </>
    );
}

SearchBar.propTypes = {
    searchBoxInput: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired
};

export default SearchBar;
