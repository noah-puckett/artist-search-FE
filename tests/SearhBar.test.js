import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from '../src/components/stateless/search/SearchBar';

describe('SearchBar component', () => {
    it('renders SearchBar', () => {
        const wrapper = shallow(<SearchBar searchBoxInput={'hello'} onChange={() => {}}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
