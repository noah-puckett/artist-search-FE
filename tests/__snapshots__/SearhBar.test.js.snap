// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders SearchBar 1`] = `
<Fragment>
  <input
    name="searchBoxInput"
    onChange={[Function]}
    value="hello"
  />
</Fragment>
`;
